reactioncommerce/reaction-next-starterkit
# ref: Update Badges to match new designs

This PR:
- updates the Badges on our Product Grid to match the styles provided in our [designs](http://zpl.io/V18RD3Q).
- introduces logic to show one or two badges, in different locations, depending on which badges are available and which pages have priority.
  - The applicable closest to top goes on the left of the product item, the next applicable goes on right side. If `Sold Out` of `Backordered`, there is never a secondary badge.
    - Sold Out or Backordered
    - Sale
    - Low Inventory
    - Bestseller

Follows the logic listed in the comment to filter out and show only certain badges on different components: https://github.com/reactioncommerce/reaction-next-starterkit/issues/88#issuecomment-394869904 - Please note that this functionality isn't fully built out, so it's mostly a placeholder as to how we should do it, but is not se in stone.

## Testing
- _Please note because of the above logic_:
  - `Sold Out` / `Backordered` badges will always be the only badge on a product, if they are applicable.
  - `Sale` badges can only be a primary badge, as the only time they _would_ be secondary is to a `Sold Out` / `Backordered` badge, which don't show secondary badges.

- Using the meteor app, add products with various levels of inventory (or add one product and adjust inventory to see changes on starterkit)
- When product is `Sold Out` or `Backordered`, you will only see one of these badge


- Inside the `renderBadge()` function of `ProductItemBadges`, hardcode `const status = { type: "SALE", label: "Sale" }` in order to fake having the sale status, as we don't have this built in yet.
  - On a product with remaining inventory, see the `Sale` badge show as the primary badge.
  - Adjust the inventory to be low inventory, and see the `Low Inventory` badge show as a secondary badge to the `Sale` badge.

- Change the hardcoded status to `const status = { type: "BEST_SELLER", label: "Best Seller" }` 
  -  `Best Seller` becomes the primary badge

- Remove the hardcoded status in the primary badge, and inside `/utils/index.js`, change `isProdcutBestSeller` to return `true`
  - See that `BestSeller` becomes a secondary badge on products with low inventory

- Add more inventory, and see that `Best Seller` becomes the primary and only badge.
